awx/main/models/credential/__init__.py

```diff
@@ -778,6 +823,11 @@ def create(self):
             'format': 'ssh_private_key',
             'secret': True,
             'multiline': True
+        }, {
+            'id': 'ssh_public_key_data',
+            'label': ugettext_noop('Signed SSH Certificate'),
+            'type': 'string',
+            'multiline': True,
```

Comment:
Is this also addressing https://github.com/ansible/awx/issues/1654?

The implementation of certificates here seems different from https://github.com/ansible/awx/issues/2434, it would be good to get a disposition on how that would be done after this is merged.

Reply:
Yes, this supports #1654 (wasn't aware that issue existed).

This implementation supports *both* use cases.

You could _either_ paste in your *signed* data _directly_ and use it *or* you can paste in *unsigned* data and associate a HashiCorp SSH Signer to it for signing purposes.  Both work.

Reply:
@AlanCoding from speaking with @wenottingham, 
https://github.com/ansible/awx/issues/2434 is a _more general_ approach to a credential type where we'd lay down a private key and a related cert (for certificate-based auth to cloud services, for example, *not* SSH).

**This** PR intends to solve the use case of "I want to do SSH certificate auth using":

-  A private key + signed certificate that _I_ generate and provide
- A private key + unsigned certificate + integration with HashiVault to sign it

Reply:
👍

Reply:
lgtm 👍 we tested signing and revocation via TTL yesterday for the second model